tests/integration/permission_client.py

```diff
@@ -1,48 +1,62 @@
-import json
+from typing import Optional
 from urllib.parse import urljoin
 
-import molgenis.client
 import requests
 
 from tests.integration.loader_mock import get_host
-from tests.integration.utils import entity_is_empty, random_name, run_commander
 
 
-def get_user_plugin_permissions(session, id_):
-    return _get_permissions(session, 'plugin', 'user', id_)
+def get_user_plugin_permission(session, plugin, user) -> Optional[str]:
+    return _get_permission(session, 'plugin', plugin, 'user', user)
 
 
-def get_user_entity_permissions(session, id_):
-    return _get_permissions(session, 'entityclass', 'user', id_)
+def get_user_entity_type_permission(session, entity_type, user) -> Optional[str]:
+    return _get_permission(session, 'entityType', entity_type, 'user', user)
 
 
-def get_user_package_permissions(session, id_):
-    return _get_permissions(session, 'package', 'user', id_)
+def get_user_package_permission(session, package, user) -> Optional[str]:
+    return _get_permission(session, 'package', package, 'user', user)
 
 
-def get_role_entity_permissions(session, id_):
-    return _get_permissions(session, 'entityclass', 'role', id_)
+def get_role_entity_type_permission(session, entity_type, role) -> Optional[str]:
+    return _get_permission(session, 'entityType', entity_type, 'role', role)
 
 
-def _get_permissions(session, resource_type, principal_type, id_):
-    # noinspection PyProtectedMember
-    headers = session._get_token_header()
-    headers.update({"Content-Type": "application/json"})
+def get_user_entity_permission(session, entity_type, entity, user) -> Optional[str]:
+    return _get_permission(session, 'entity-' + entity_type, entity, 'user', user)
+
+
+def get_role_entity_permission(session, entity_type, entity, role) -> Optional[str]:
+    return _get_permission(session, 'entity-' + entity_type, entity, 'role', role)
+
+
+def _get_permission(session, type_id, object_id, principal_type, principal) -> Optional[str]:
+    headers = _configure_headers(session)
     response = requests.get(urljoin(get_host()['url'],
-                                    'menu/admin/permissionmanager/{}/{}/{}'.format(resource_type, principal_type, id_)),
+                                    'api/permissions/{}/{}?q={}=={}'.format(type_id,
+                                                                            object_id,
+                                                                            principal_type,
+                                                                            principal)),
```

Comment:
It's hard to decide what to trust and what not when writing these tests. This feels like it should be trusted code in the permissions api client, used in integration tests. Then in unit tests you could verify that the implementation calls the right endpoints.